### PR TITLE
Feature #213 solidify source badges

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Feature #213 solidify source badges
https://github.com/unfoldingWord-dev/door43.org/issues/213

Changes:
- a couple minor changes to Door43Deployer to enable unit testing.
- added unit tests and test data for Door43Deployer.
- disabled TestDeployToDoor43() since it is calling undefined method.

### CODE REVIEW

File: functions/deploy/main.py [hunk 1]
```diff
@@ -26,8 +26,14 @@ class Door43Deployer(object):
     def __init__(self, cdn_bucket, door43_bucket):
         self.cdn_bucket = cdn_bucket
         self.door43_bucket = door43_bucket
-        self.cdn_handler = S3Handler(cdn_bucket)
-        self.door43_handler = S3Handler(door43_bucket)
+
+        self.cdn_handler = S3Handler(cdn_bucket) if cdn_bucket else None
+        self.door43_handler = S3Handler(door43_bucket) if door43_bucket else None
```
Comment (phillip-hopper) [inline functions/deploy/main.py:31]: What is the purpose of this change? The handlers are required, so this seems to just ignore an error.

File: functions/deploy/main.py [hunk 2]
```diff
@@ -26,8 +26,14 @@ class Door43Deployer(object):
...
+
+        # keep track of temp folders for easier cleanup
+        self.source_dir = None
+        self.output_dir = None
+        self.template_dir = None
```
Comment (phillip-hopper) [inline functions/deploy/main.py:36]: Has cleanup code been added also? If not, this seems to do nothing.

File: functions/deploy/main.py [hunk 3]
```diff
@@ -52,6 +58,11 @@ def deploy_commit_to_door43(self, build_log_key):
         output_dir = tempfile.mkdtemp(prefix='output_')
         template_dir = tempfile.mkdtemp(prefix='template_')
 
+        # keep track of temp folders for easier cleanup
+        self.source_dir = source_dir
+        self.output_dir = output_dir
+        self.template_dir = template_dir
```
Comment (phillip-hopper) [inline functions/deploy/main.py:64]: Ditto.

File: tests/door43_deployer/test_door43_deployer.py [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+    #
+    #     #given
+    #     resource = 'converted'
+    #     repo_name = 'aai_obs_text_obs-missing_chapter_50_old'
+    #
+    #     # when
+    #     zip_file = self.packageResource(resource, repo_name)
+    #
+    #     #then
+    #     print(zip_file)
```
Comment (phillip-hopper) [inline tests/door43_deployer/test_door43_deployer.py:256]: What is the purpose of leaving these lines in the file?

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

File: functions/deploy/main.py (hunk 1 above)
Comment (PhotoNomad0) [reply functions/deploy/main.py:31]: @phillip-hopper - as written the method was not very easy to unit test.  I was trying to make minimal changes.  I was trying to work around exceptions being thrown in init method.  I will look for another approach.

File: functions/deploy/main.py (hunk 3 above)
Comment (PhotoNomad0) [reply functions/deploy/main.py:64]: @phillip-hopper - These changes were for unit testing cleanup.  I probably should just add cleanup of temp folders to deploy method.

File: functions/deploy/main.py (hunk 1 above)
Comment (phillip-hopper) [reply functions/deploy/main.py:31]: @PhotoNomad0 We started passing the S3Handler class to the functions so it was easy to mock during testing. See https://github.com/unfoldingWord-dev/d43-catalog/blob/develop/tests/signing/test_signing.py.

**Comment by richmahn** [comment]
The S3 and dynomodb have already been handled in the merge with tx-manager repo. All lambda functions pass classes for dealing with resources.

File: functions/deploy/main.py (hunk 1 above)
Comment (PhotoNomad0) [reply functions/deploy/main.py:31]: @phillip-hopper - thanks, will take a look as I have time.
